api/src/org/apache/cloudstack/api/command/admin/network/DeleteManagementNetworkIpRangeCmd.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.cloudstack.api.ServerApiException;
+import org.apache.cloudstack.api.response.SuccessResponse;
+
+import com.cloud.event.EventTypes;
+import com.cloud.exception.ConcurrentOperationException;
+import com.cloud.exception.InsufficientCapacityException;
+import com.cloud.exception.ResourceAllocationException;
+import com.cloud.exception.ResourceUnavailableException;
+import com.cloud.user.Account;
+
+@APICommand(name = "deleteManagementNetworkIpRange",
+        description = "Deletes a management network IP range. This action is only allowed when no IPs in this range are allocated.",
+        responseObject = SuccessResponse.class,
+        since = "4.10.0.0",
+        requestHasSensitiveInfo = false,
+        responseHasSensitiveInfo = false)
+public class DeleteManagementNetworkIpRangeCmd extends BaseAsyncCmd {
+    public static final Logger s_logger = Logger.getLogger(DeleteManagementNetworkIpRangeCmd.class);
+
+    private static final String s_name = "deletemanagementnetworkiprangeresponse";
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+
+    @Parameter(name = ApiConstants.POD_ID,
+            type = CommandType.UUID,
+            entityType = PodResponse.class,
+            required = true,
+            description = "UUID of POD, where the IP range belongs to.")
+    private Long podId;
+
+    @Parameter(name = ApiConstants.START_IP,
+            type = CommandType.STRING,
+            required = true,
+            description = "The starting IP address.")
+    private String startIp;
+
+    @Parameter(name = ApiConstants.END_IP,
+            type = CommandType.STRING,
+            required = true,
+            description = "The ending IP address.")
+    private String endIp;
+
+    /////////////////////////////////////////////////////
+    /////////////////// Accessors ///////////////////////
+    /////////////////////////////////////////////////////
+
+    public Long getPodId() {
+        return podId;
+    }
+
+    public String getStartIp() {
+        return startIp;
+    }
+
+    public String getEndIp() {
+        return endIp;
+    }
+
+    @Override
+    public String getEventType() {
+        return EventTypes.EVENT_MANAGEMENT_IP_RANGE_DELETE;
+    }
+
+    @Override
+    public String getEventDescription() {
+        return "Deleting management ip range from " + getStartIp() + " to " + getEndIp() + " of Pod: " + getPodId();
+    }
+
+    @Override
+    public void execute() throws ResourceUnavailableException, InsufficientCapacityException, ServerApiException, ConcurrentOperationException,
+            ResourceAllocationException {
+        try {
+            _configService.deletePodIpRange(this);
+            SuccessResponse response = new SuccessResponse(getCommandName());
+            this.setResponseObject(response);
+        } catch (Exception e) {
```

Comment:
I would like more specific catch clauses. catch'm'all is a bit crude (and it will also happen on the ApiServlet level.

Reply:
I will modify this.